models/ssh_key.go

```diff
@@ -361,18 +361,11 @@ func checkKeyFingerprint(e Engine, fingerprint string) error {
 
 func calcFingerprint(publicKeyContent string) (string, error) {
 	// Calculate fingerprint.
-	tmpPath, err := writeTmpKeyFile(publicKeyContent)
+	pk, _, _, _, err := ssh.ParseAuthorizedKey([]byte(publicKeyContent))
 	if err != nil {
 		return "", err
 	}
-	defer os.Remove(tmpPath)
-	stdout, stderr, err := process.GetManager().Exec("AddPublicKey", "ssh-keygen", "-lf", tmpPath)
-	if err != nil {
-		return "", fmt.Errorf("'ssh-keygen -lf %s' failed with error '%s': %s", tmpPath, err, stderr)
-	} else if len(stdout) < 2 {
-		return "", errors.New("not enough output for calculating fingerprint: " + stdout)
-	}
-	return strings.Split(stdout, " ")[1], nil
+	return ssh.FingerprintSHA256(pk), nil
```

Comment:
This seems to be an unrelated change. Can we move the change into a new PR?

Reply:
It is a needed change because openssh does un-needed system check (that fail in this configuration with --user flag) for calculating the fingerprint of the key. I allready separate this in an other PR #7128

Reply:
What system check is failing for you?

I setup a test with alpine and the `--user` flag, but can not see any error.

https://gist.github.com/das7pad/62b58bfb67da516542992d8bf5744446

Reply:
If you pass an uid/gid (numerical) that doesn't exist inside the container it should failed.